src/components/NewBook.js

```diff
@@ -1,26 +1,50 @@
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+// import PropTypes from 'prop-types';
 import styles from '../styles/NewBook.module.scss';
+import { addBook } from '../redux/books/booksSlice';
 
-const NewBook = () => (
-  <div>
-    <h2>Add New Book</h2>
-    <form className={styles.form}>
-      <input type="text" placeholder="Book title" />
-      <select name="languages" id="lang">
-        <option value="" disabled>
-          Category
-        </option>
-        <option value="javascript">JavaScript</option>
-        <option value="php">PHP</option>
-        <option value="java">Java</option>
-        <option value="golang">Golang</option>
-        <option value="python">Python</option>
-        <option value="c#">C#</option>
-        <option value="C++">C++</option>
-        <option value="erlang">Erlang</option>
-      </select>
-      <input type="submit" value="Add New Book" />
-    </form>
-  </div>
-);
+const NewBook = () => {
+  const val = Math.random() * 1000;
+  const [newBook, setNewBook] = useState({
+    item_id: val.toString(),
+    title: '',
+    author: '',
+    category: 'Nonfiction',
+  });
+  const dispatch = useDispatch();
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(addBook(newBook));
+  };
+  return (
+    <div>
+      <h2>Add New Book</h2>
+      <form className={styles.form} onSubmit={handleSubmit}>
+        <input
+          type="text"
+          defaultValue="title"
+          placeholder="Book title"
+          onChange={(e) => setNewBook({ ...newBook, title: e.target.value })}
+        />
```

Comment:
- [ Required ] I think it would be better if you remove the `default value` from the `title` input field to give your users a good user experience when filling out the form. Currently, I have to delete the text in the input field before filling in the value of the title which is not a good user experience.